### ISSUE TITLE
Consider removing <all_urls> permission

### ISSUE THREAD

**Issue by bershanskiy** [mozilla/fx-private-relay#252]
## Summary
As a user, I avoid extensions that require permission to "Access your data for all websites". As far as I see, this permission can be either removed entirely or made optional.

## Details
As far as I see, the extension is activated when user interacts with it either via context menu or the extension icon. If so, the access to the current page is granted upon interaction and thus the explicit warning-inducing `<all_urls>` permission is unnecessary.

Would you be interested in reviewing a PR if I prepared it?

**Comment by bershanskiy** [comment]
After a bit of testing, I see that this extension adds a button to email fields in forms, which requires `<all_urls>` permission. However, I still prefer at least an option to disable this functionality and not have extension running on every site (run it only when I explicitly activate it).

**Comment by groovecoder** [comment]
Thanks for filing the issue. We could consider [requesting the permission at run-time](https://extensionworkshop.com/documentation/develop/request-the-right-permissions/#request-permissions-at-runtime) as part of an future on-boarding experience that introduces the button.

**Comment by bershanskiy** [comment]
@groovecoder Yes, I'd prefer an optional permission requested during on-boarding. [Content scripts API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/contentScripts) might be more convenient here since extension needs to register a whole script (not just get a permission for an existing script).
Would you be interested in reviewing a patch if I prepared it? It should be a rather small and trivial change (on-boarding UI mostly).

**Comment by groovecoder** [comment]
We'll wait for our UX folks to put together any on-boarding for us. Thanks though.
